sympy/geometry/polygon.py

```diff
@@ -783,6 +783,59 @@ def intersection(self, o):
             return list(ordered(intersection_result))
 
 
+    def cut_section(self, line):
```

Comment:
I think this algorithm will only work for a convex polygon. If there are concave regions you would have to replace all connected points below.
```python
>>> ice = Polygon((0,0),(1,2.5),(2,1),(3,2.5),(4,2),(5,3),(-1,3))
>>> ice.cut_section(Line((0,0),(4.5,3)))
Polygon(Point2D(0, 0), Point2D(9/2, 3), Point2D(1, 5/2), Point2D(3, 5/2), Point2D(-1, 3))
```
but it should be
```python
>>> ans
Polygon(
 Point2D(0, 0), Point2D(1, 5/2), Point2D(24/13, 16/13), Point2D(12/5, 8/5),
 Point2D(3, 5/2), Point2D(24/7, 16/7), Point2D(9/2, 3), Point2D(-1, 3))
```

Reply:
@smichr I have made some changes to make it handle concave polygons, please have a look.
The above example gives the correct result now.